src/invocation/InvocationService.ts

```diff
@@ -106,19 +124,35 @@ export class InvocationService {
         var messageType = clientMessage.getMessageType();
 
         if (clientMessage.hasFlags(BitsUtil.LISTENER_FLAG)) {
-            Q.fcall(this.eventHandlers[correlationId].handler, clientMessage);
```

Comment:
I put there a fcall because it puts the handler at the end of the event loop. If we call the handler directly, it is called immediately. We should break execution logic into parts as best as we can. You may use `setImmediate` instead of `Q.fcall` if unused promise bothers you.